design/media_types.go

```diff
@@ -30,6 +30,7 @@ var workItem = a.MediaType("application/vnd.workitem+json", func() {
 	a.Description("A work item hold field values according to a given field type")
 	a.Attribute("id", d.String, "unique id per installation")
 	a.Attribute("version", d.Integer, "Version for optimistic concurrency control")
+	a.Attribute("ExecutionOrder", d.Number, "Order of the workitem in the list")
```

Comment:
Keep in mind there are 3 objects in play here

app.WorkItem2.Data.Attributes[system.Order]
app.WorkItem.Fields[system.Order]
workitem.WorkItem.ExecutionOrder